e2e/tenv.ts

```diff
@@ -0,0 +1,126 @@
+/* eslint-disable import/no-nodejs-modules */
+import childProcess from 'child_process';
+import fs from 'fs/promises';
+import { createServer } from 'http';
+import { AddressInfo } from 'net';
+import path from 'path';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { Repeater } from '@repeaterjs/repeater';
+
+let leftovers: Proc[] = [];
+afterAll(async () => {
+  await Promise.allSettled(
+    leftovers.map(proc => {
+      proc.kill();
+      return proc.waitForExit;
+    }),
+  );
+  leftovers = [];
+});
+
+export interface Proc {
+  stdout: Repeater<string>;
+  stderr: Repeater<string>;
+  kill(code?: number): void;
+  waitForExit: Promise<void>;
+}
+
+export interface Tenv {
+  getAvailablePort(): number;
+  spawn(cmd: string, ...args: (string | number)[]): Promise<Proc>;
+  fileExists(filePath: string): Promise<boolean>;
+  readFile(filePath: string): Promise<string>;
+  // If the file doesn't exist, is a noop.
+  deleteFile(filePath: string): Promise<void>;
+}
+
+export function createTenv(cwd: string): Tenv {
```

Comment:
It could be interesting, when creating a testing env, to create a temp directory and use it as the working directory. This would simply the cleanup :-)

Reply:
Working directory should be the one that has Mesh installed and contains the configurations.